contracts/margin/external/BucketLender/BucketLender.sol

```diff
@@ -0,0 +1,699 @@
+
+        /* CHECK VALUES256 */
+        // no need to require anything about loanOffering.maximumAmount
+        // no need to require anything about loanOffering.minimumAmount
+        // no need to require anything about loanOffering.minimumHeldToken
+        require(loanOffering.lenderFee == 0);
+        // no need to require anything about loanOffering.takerFee
+        // no need to require anything about loanOffering.expirationTimestamp
+        // no need to require anything about loanOffering.salt
+
+        /* CHECK VALUES32 */
+        // no need to require anything about loanOffering.callTimeLimit
+        // no need to require anything about loanOffering.maxDuration
+        // no need to require anything about loanOffering.interestRate
+        // no need to require anything about loanOffering.interestPeriod
+
+        return address(this);
+    }
+
+    /**
+     * Called by the Margin contract when anyone transfers ownership of a loan to this contract.
+     * This function initializes this contract and returns this address to indicate to Margin
+     * that it is willing to take ownership of the loan.
+     *
+     * @param  from        (unused)
+     * @param  positionId  Unique ID of the position
+     * @return             This address on success, throw otherwise
+     */
+    function receiveLoanOwnership(
+        address from,
+        bytes32 positionId
+    )
+        external
+        onlyMargin
+        onlyPosition(positionId)
+        returns (address)
+    {
+        MarginCommon.Position memory position = MarginHelper.getPosition(DYDX_MARGIN, POSITION_ID);
+
+        assert(position.principal > 0);
+        assert(position.owedToken == OWED_TOKEN);
+        assert(position.heldToken == HELD_TOKEN);
+
+        // set relevant constants
+        uint256 initialPrincipal = position.principal;
+        principal.bucket[0] = initialPrincipal;
+        principal.total = initialPrincipal;
+        weight[0].total = weight[0].total.add(initialPrincipal);
+        weight[0].account[from] = weight[0].account[from].add(initialPrincipal);
+
+        return address(this);
+    }
+
+    /**
+     * Called by Margin when additional value is added onto the position this contract
+     * is lending for. Balance is added to the address that loaned the additional tokens.
+     *
+     * @param  payer           Address that loaned the additional tokens
+     * @param  positionId      Unique ID of the position
+     * @param  principalAdded  Amount that was added to the position
+     *  param  lentAmount      (unused)
+     * @return                 This address to accept, a different address to ask that contract
+     */
+    function increaseLoanOnBehalfOf(
+        address payer,
+        bytes32 positionId,
+        uint256 principalAdded,
+        uint256 lentAmount
+    )
+        external
+        onlyMargin
+        onlyPosition(positionId)
+        returns (address)
+    {
+        assert(payer == address(this));
```

Comment:
should be `require`

Reply:
require could happen in the case of `increaseWithoutCounterparty` or just if some random was attempted to be used to increase the position

Reply:
(should test both)